File: app/src/main/java/lt/tesonet/ui/servers/ServersAdapter.java
package lt.tesonet.ui.servers;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import lt.tesonet.R;
import lt.tesonet.model.Server;

public class ServersAdapter extends RecyclerView.Adapter<ServersAdapter.ViewHolder> {

    private List<Server> data = new ArrayList<>();

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int position) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_server, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int position) {
        Context context = viewHolder.itemView.getContext();

        Server server = data.get(position);
        viewHolder.name.setText(server.getName());
        viewHolder.distance.setText(context.getString(R.string.server_distance_pattern, server.getDistance()));
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public void setData(List<Server> servers) {
        data.clear();
        data.addAll(servers);
    }

    static class ViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.name)
        TextView name;
        @BindView(R.id.distance)
        TextView distance;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/kotlin/com/ne2rnas/party/ui/servers/ServersAdapter.kt
package com.ne2rnas.party.ui.servers

import android.content.Context
import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.ne2rnas.party.R
import com.ne2rnas.party.data.servers.Server
import com.ne2rnas.party.databinding.ItemServerBinding

class ServersAdapter(private val context: Context) : RecyclerView.Adapter<ServersAdapter.ServerViewHolder>() {

    private var servers: List<Server> = listOf()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ServerViewHolder {
        val layoutInflater = LayoutInflater.from(context)
        val binding: ItemServerBinding = DataBindingUtil.inflate(layoutInflater, R.layout.item_server, parent, false)
        return ServerViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ServerViewHolder, position: Int) {
        holder.bind(servers[position])
    }

    override fun getItemCount(): Int {
        return servers.size
    }

    fun updateServers(servers: List<Server>) {
        this.servers = servers
        notifyDataSetChanged()
    }

    class ServerViewHolder(private val binding: ItemServerBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(server: Server) {
            binding.server = server
            binding.executePendingBindings()
        }
    }
}
